Apply changes from filter chip dropdowns immediately

emma-sg opened the pull request:
## Changes

This PR addresses some user feedback about filter chip dropdowns being confusing: specifically, about how when selecting a filter from a dropdown, it's not clear that you need to click outside the dropdown to apply the changes, leading to worries about user actions not being saved (see [source in Discord](https://discord.com/channels/895426029194207262/1423361788589637686/1423362488636014705)).

This addresses this by changing filter chip behaviour to update its state (and re-fetch data) immediately when changes are made inside the dropdowns.

This also reworks the "schedule" filter chip in the workflow list page to show the selected option with a radio button so that it's more clear which state is selected and how switching states will work.

Finally, this addresses a bug in the archived item list where changing filters repeatedly would cause cascades of unnecessary re-fetches.

## Demo

https://github.com/user-attachments/assets/51bac346-0776-4384-afab-9604f9289b50

## Testing

On the workflow & archived item pages, try changing all of the filters, and make sure they all update & can be cleared (where relevant) as expected.

emma-sg committed:
rework filter chips so that all of them apply their changes immediately

this resolves some user confusion around needing to close the dropdown
before changes would be applied

emma-sg committed:
ensure timeout is cancelled when fetching archived items

this fixes an issue where every time filters were changed, another
infinitely-repeating timeout was created, leading to long cascades of
fetches in some cases.

emma-sg committed:
fix broken clear button on archived item tag filter dropdown

emma-sg committed:
fix type switcher not dispatching update events in workflow tag filter

emma-sg committed:
pass workflow tag filter type into tag filter

emma-sg committed:
simplify & fix clear button issues on workflow tag & profile filters

SuaYoo commented on frontend/src/features/archived-items/archived-item-state-filter.ts:
```diff
@@ -69,6 +69,15 @@ export class ArchivedItemStateFilter extends BtrixElement {
+        }),
+      );
+    }
   }
```
Prefer to have the event dispatched in the updated callback, since `willUpdate` should generally be used to compute and batch updates and `updated` better indicates that the value has been committed.

```suggestion
  }

  protected updated(changedProperties: PropertyValues<this>): void {
    if (changedProperties.has("selectedStates")) {
      this.dispatchEvent(
        new CustomEvent<
          BtrixChangeEvent<ChangeArchivedItemStateEventDetails>["detail"]
        >("btrix-change", {
          detail: { value: this.selectedStates },
        }),
      );
    }
  }
```

emma-sg replied:
Sure, makes sense!

emma-sg committed:
remove redundant `selectedProfiles` state and use `selected` directly in
workflow profile filter

emma-sg committed:
fix check for wrong changed property

SuaYoo commented on frontend/src/features/crawl-workflows/workflow-profile-filter.ts:
```diff
@@ -344,6 +338,7 @@ export class WorkflowProfileFilter extends BtrixElement {
           const { checked, value } = e.target as SlCheckbox;
 
           this.selected.set(value, checked);
+          this.requestUpdate("selected");
```
This and the other calls to `requestUpdate` can be removed with the `selected` state change.

```suggestion
          this.selected = new Map(this.selected.set(value, checked));
```

emma-sg committed:
apply suggestions from code review

- remove redundant computed properties used as state, and instead track
  selected maps
- replace `requestUpdate` calls with assignment to new `Map`s
- use `updated` instead of `willUpdate` to dispatch events

Co-authored-by: sua yoo <sua@suayoo.com>

emma-sg committed:
remove missed `requestUpdate`

emma-sg committed:
remove unused requestUpdate

emma-sg committed:
fix hangs and clean up unused property on archived item tag filter

emma-sg committed:
fix archived item tag filter type change not dispatching event

- also fixes hang
